Carry #18912 : Docker Remote API documentation update

vdemeester opened the pull request:
Carry #18912. Taken into account @jamtur01 and @moxiegirl comments.

🐸

/cc @jamtur01 @thaJeztah @moxiegirl @MHBauer 

---

I was confused for the longest time on how to actually use and make requests against the remote API, so I think that it might help for those getting started with it to know how to actually test it out via curl. I added in parts on how to access the remote API via curl against the default unix socket, and also on how to configure the docker daemon to expose the API on a TCP port as well if desired.

Signed-off-by: Michelle Liu michelle@michelleliu.io

MHBauer commented:
Something weird happened I think. It looks like the first paragraph is repeated.

vdemeester commented:
@MHBauer oh you're right, my bad :angel:

MHBauer commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -24,6 +24,20 @@ client must have `root` access to interact with the daemon. If a group named
+
+    curl --unix-socket /var/run/docker.sock http://containers/json
+
+If you bound the Docker daemon to a different, you would reference that
```
bound to a different _what_? Port doesn't really make sense here. I don't think unix sockets have ports.

MHBauer commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -24,6 +24,20 @@ client must have `root` access to interact with the daemon. If a group named
+alternative port in your cURL rather than the default.
+
+Please note that exposing the Remote API can be a security issue and
+you should consider this before binding the API.
```
Seems random to have this warning here. I agree a warning is needed somewhere. I'm not sure where.

vdemeester replied:
You're right. Pretty sure it is somewhere else already even.

vdemeester commented:
@MHBauer updated.

thaJeztah commented:
LGTM

icecrime commented:
Ping @moxiegirl @MHBauer!

tiborvass commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
 `docker` exists on your system, `docker` applies ownership of the socket to the
 group.
 
+To interact with the remote API using [cURL](http://curl.haxx.se/download.html).
```
`https`

vdemeester replied:
oh nooeess :scream:

vdemeester commented:
@tiborvass updated :wink:

tiborvass commented:
LGTM

tiborvass commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
+
+    curl --unix-socket /var/run/docker.sock http://containers/json
+
+If you bound the Docker daemon to a different socket path or TCP port,
```
shouldnt this be `bind`?

jamtur01 replied:
If you have bound the ...

tiborvass commented:
Need a native-speaker LGTM @MHBauer @jamtur01

jamtur01 commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
 `docker` exists on your system, `docker` applies ownership of the socket to the
 group.
 
+To interact with the remote API using [cURL](https://curl.haxx.se/download.html).
```
This is either a sentence fragment or a header? Unclear.

jamtur01 commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
 
+To interact with the remote API using [cURL](https://curl.haxx.se/download.html).
+You run `curl` against the socket where the Docker daemon is listening. For this
+reason, must use cURL 7.40 (or greater) as these versions have the
```
To connect to the Docker daemon with `cURL` you need to use version 7.40 or later, as these versions provide the `--unix-socket` flag.

jamtur01 commented on docs/reference/api/docker_remote_api.md:
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
+    curl --unix-socket /var/run/docker.sock http://containers/json
+
+If you bound the Docker daemon to a different socket path or TCP port,
+you would reference that later in your cURL rather than the default.
```
-later

jamtur01 commented:
Needs some rewrites I think.

anonmily committed:
Docker Remote API documentation update

I was confused for the longest time on how to actually use and make requests against the remote API, so I think that it might help for those getting started with it to know how to actually test it out via curl. I added in parts on how to access the remote API via curl against the default unix socket, and also on how to configure the docker daemon to expose the API on a TCP port as well if desired.

Signed-off-by: Michelle Liu <michelle@michelleliu.io>

vdemeester commented:
updated ^_^

thaJeztah commented:
thanks, still LGTM
